crates/subspace-solving/src/codec.rs

```diff
@@ -133,16 +134,30 @@ impl SubspaceCodec {
 
         #[cfg(feature = "cuda")]
         if self.cuda_available {
-            let pieces_to_process = pieces.len() / PIECE_SIZE / GPU_PIECE_BLOCK * GPU_PIECE_BLOCK;
-
-            self.batch_encode_cuda(
-                &mut pieces[..pieces_to_process * PIECE_SIZE],
-                &piece_indexes[..pieces_to_process],
-            )?;
-
-            // Leave the rest for CPU
-            pieces = &mut pieces[pieces_to_process * PIECE_SIZE..];
-            piece_indexes = &piece_indexes[pieces_to_process..];
+            let mut pieces_to_process = pieces.len() / PIECE_SIZE;
+
+            // GPU will accept multiples of 1024 pieces
+            if pieces_to_process >= 1024 {
```

Comment:
the most important change is here

Reply:
There is a `GPU_PIECE_BLOCK` constant above, please use it instead, then you don't even need a comment here